src/shared/configurationProvider.ts

```diff
@@ -226,58 +233,69 @@ export class BaseVsDbgConfigurationProvider implements vscode.DebugConfiguration
         }
     }
 
-    private checkForDevCerts(dotnetPath: string) {
-        hasDotnetDevCertsHttps(dotnetPath).then(async (returnData) => {
+    private async checkForDevCerts(dotnetPath: string) {
+        await hasDotnetDevCertsHttps(dotnetPath).then(async (returnData) => {
             const errorCode = returnData.error?.code;
             if (
                 errorCode === CertToolStatusCodes.CertificateNotTrusted ||
                 errorCode === CertToolStatusCodes.ErrorNoValidCertificateFound
             ) {
-                const labelYes = vscode.l10n.t('Yes');
+                const labelYes: ActionOption = {
+                    title: vscode.l10n.t('Yes'),
+                    action: async () => {
+                        const returnData = await createSelfSignedCert(dotnetPath);
+                        if (returnData.error === null) {
+                            //if the prcess returns 0, returnData.error is null, otherwise the return code can be acessed in returnData.error.code
```

Comment:
```suggestion
                            // if the prpcess returns 0, returnData.error is null, otherwise the return code can be acessed in returnData.error.code
```